spikeinterface/preprocessing/normalize_scale.py

```diff
@@ -215,14 +227,31 @@ class ZScoreRecording(BasePreprocessor):
     """
     name = 'center'
 
-    def __init__(self, recording, mode="median+mad",
+    def __init__(self, recording, mode="median+mad", gain=None, offset=None, int_scale=None,
                  dtype="float32", **random_chunk_kwargs):
         
         assert mode in ("median+mad", "mean+std")
 
+        # fix dtype
+        dtype_ = fix_dtype(recording, dtype)
+
+        if dtype_.kind == 'i':
+            assert int_scale is not None, 'For recording with dtype=int you must set dtype=float32 OR set a scale' 
```

Comment:
If user sets dtype=float32 here, rather than scaling, and runs the sorter, isn't the recording coing to be silently cast back to integer in `write_binary_recording`?